scripts/cmake-build.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2025 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -uo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+version="${CMAKE_VERSION:="3.31.6"}"
+target_dir="${TARGET_DIRECTORY:=""}"
+
+if [ -z "$version" ]; then
+  fatal "CMake version must be specified."
+fi
+if [ -z "$target_dir" ]; then
+  fatal "Target directory must be specified."
+fi
+
+log "Installing tools for this script"
+apt-get update -y -q && apt-get install -y -q curl rsync
+
+log "Obtaining CMake (${CMAKE_VERSION})"
+curl -sL "https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz" -o /tmp/cmake.tar.gz
```

Comment:
Is there a specific need to pin to the CMake version? If not, we could just get this from `apt` like the other packages?

Reply:
I don't think the pinning is necessary from our side, but it doesn't seem to be available in `apt` by default nor is there an 'official' repository I could add to pull it from that I could see.

Reply:
Huh - strange, I'm not sure why my attempt to find it failed. I'll use this one